app/src/main/java/com/peopleofandroido/chillaxingcat/presentation/viewmodel/UserSettingViewModel.kt

```diff
@@ -0,0 +1,267 @@
+        Event(Action())
+    )
+    val actionEvent: NotNullMutableLiveData<Event<Action>>
+        get() = _actionEvent
+
+    private val _reminderTime: NotNullMutableLiveData<String> = NotNullMutableLiveData("18:00")
+    val reminderTime: NotNullMutableLiveData<String>
+        get() = _reminderTime
+
+    private val _reminderText: NotNullMutableLiveData<String> = NotNullMutableLiveData("")
+    val reminderText: NotNullMutableLiveData<String>
+        get() = _reminderText
+
+    private val _goalRestingTimeHour: NotNullMutableLiveData<String> = NotNullMutableLiveData("")
+    val goalRestingTimeHour: NotNullMutableLiveData<String>
+        get() = _goalRestingTimeHour
+    private val _goalRestingTimeMinute: NotNullMutableLiveData<String> = NotNullMutableLiveData("")
+    val goalRestingTimeMinute: NotNullMutableLiveData<String>
+        get() = _goalRestingTimeMinute
+
+    private val _pushEnable: NotNullMutableLiveData<Boolean> = NotNullMutableLiveData(true)
+    val pushAvailable: NotNullMutableLiveData<Boolean>
+        get() = _pushEnable
+    private var mCalender: GregorianCalendar? = null
+    var isInitial: Boolean = false
+
+    private var alarmManager: AlarmManager?
+
+    init {
+        mCalender = GregorianCalendar()
+        alarmManager =
+            getApplication<Application>().getSystemService(Context.ALARM_SERVICE) as AlarmManager?
+        loadPushAvailable()
+        loadReminderTime()
+        loadReminderText()
+        loadGoalRestingTime()
+    }
+
+    private fun loadPushAvailable() {
+        viewModelScope.launch() {
+            val result = useCases.getNotificationStatus()
+            result.data?.let { it ->
+                pushAvailable.value = it
+            }
+            logd("getNotificationStatus()::data: ${result.data}, message: ${result.message}, status: ${result.status}")
+        }
+    }
+
+    private fun loadReminderTime() {
+        viewModelScope.launch() {
+            val result = useCases.getReminderTime()
+            result.data?.let { it ->
+                if (it.isNotEmpty()) {
+                    reminderTime.value = it
+                }
+            }
+            logd("getReminderTime()::data: ${result.data}, message: ${result.message}, status: ${result.status}")
+        }
+    }
+
+    private fun loadReminderText() {
+        viewModelScope.launch() {
+            val result = useCases.getReminderText()
+            result.data?.let { it ->
+                if (it.isNotEmpty()) {
+                    reminderText.value = it
+                }
+            }
+            logd("getReminderText()::data: ${result.data}, message: ${result.message}, status: ${result.status}")
+        }
+    }
+
+    private fun loadGoalRestingTime() {
+        viewModelScope.launch() {
+            val result = useCases.getGoalRestingTime()
+            result.data?.let { it ->
+                if (it.isNotEmpty() && it.contains(":")) {
```

Comment:
```suggestion
        val regex = Regex("[0-2][0-9][:][0-5][0-9]")
	if (regex.matchEntire(it)?.groupValues?.size?:0 > 0) {
```

문제는 없으나.. 좀 더 개선해보자면 이 코드는 어떨지 제안해봅니다.ㅎ